### PR TITLE
doc(aiplatform): add Vertex AI samples

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14703)

### CODE REVIEW

File: google/cloud/aiplatform/samples/samples.cc [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    }
+  }
+  // [END generativeaionvertexai_gemini_generate_from_text_input]
+  (argv.at(0), argv.at(1), argv.at(2), {argv.begin() + 3, argv.end()});
+}
+
+void GeminiGenerateWithImage(std::vector<std::string> const& argv) {
+  if (argv.size() != 6) {
+    throw google::cloud::testing_util::Usage(
+        "gemini-generate-from-text-input <project> <location> <model-name> "
```
Comment (dbolduc) [inline google/cloud/aiplatform/samples/samples.cc:63]: `gemini-generate-with-image`

File: google/cloud/aiplatform/samples/samples.cc [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    }
+  }
+  // [END generativeaionvertexai_gemini_get_started]
+  (argv.at(0), argv.at(1), argv.at(2), argv.at(3), argv.at(4), argv.at(5));
+}
+
+void GeminiVideoWithAudio(std::vector<std::string> const& argv) {
+  if (argv.size() != 6) {
+    throw google::cloud::testing_util::Usage(
+        "gemini-generate-from-text-input <project> <location> <model-name> "
```
Comment (dbolduc) [inline google/cloud/aiplatform/samples/samples.cc:108]: `gemini-video-with-audio`

File: google/cloud/aiplatform/samples/samples.cc [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+  std::cout << "Executing GeminiGenerateFromTextInput sample:\n";
+  GeminiGenerateFromTextInput(
+      {project_id, "us-central1", "gemini-1.5-flash-001",
+       "What's a good name for a flower shop that specializes in selling "
+       "bouquets of dried flowers?"});
+
+  std::cout << "\nExecuting GeminiGenerateWithImage sample:\n";
+  GeminiGenerateWithImage({project_id, "us-central1", "gemini-1.5-flash-001",
+                           "What's in this photo?", "image/png",
+                           "gs://generativeai-downloads/images/scones.jpg"});
```
Comment (dbolduc) [inline google/cloud/aiplatform/samples/samples.cc:162]: I am going to guess: scones

**Review by dbolduc — APPROVED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14703)* status: 0 of 4 files reviewed, 3 unresolved discussions (waiting on @dbolduc)

___
*[`google/cloud/aiplatform/samples/samples.cc` line 63 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14703#gh-1759296100:-O6gjAJ41qRc-9dg-quq:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/fb189ceb33ddd1c8cd308e5b5249c8aedf9d788d/google/cloud/aiplatform/samples/samples.cc#L63)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

`gemini-generate-with-image`
</blockquote></details>

Done.
___
*[`google/cloud/aiplatform/samples/samples.cc` line 108 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14703#gh-1759296334:-O6gjAqW0ezmhCJI7v4r:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/fb189ceb33ddd1c8cd308e5b5249c8aedf9d788d/google/cloud/aiplatform/samples/samples.cc#L108)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

`gemini-video-with-audio`
</blockquote></details>

Done.
___
*[`google/cloud/aiplatform/samples/samples.cc` line 162 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14703#gh-1759297072:-O6gjB8n0k63j4mSa2O3:bktc8sb) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/fb189ceb33ddd1c8cd308e5b5249c8aedf9d788d/google/cloud/aiplatform/samples/samples.cc#L162)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

I am going to guess: scones
</blockquote></details>

The photo shows a table setting with a white tablecloth, blueberry scones, a bowl of blueberries, two cups of coffee, and pink peonies. The spoon says "Let's Jam".

**Pull request merged**